holobot/sdk/database/queries/on_conflict_update_builder.py

```diff
@@ -37,35 +37,31 @@ def build(self) -> tuple[str, tuple[Any, ...]]:
         if not self.__fields:
             raise ValueError("The UPDATE clause must have at least one field.")
 
-        parent_sql, parent_arguments = self.__parent_builder.build()
-        arguments = list(parent_arguments)
+        parent_sql, arguments = self.__parent_builder.build()
         sql = [parent_sql, "UPDATE", self.__table_name, "SET"]
-        columns = []
-        argument_count = len(arguments)
+        arguments = []
         if len(self.__fields) > 1:
-            sql.append("(")
-            for column in self.__fields.keys():
-                columns.append(column)
-            sql.append(", ".join(columns))
-            sql.append(") = (")
+            sql.extend(("(", ", ".join(self.__fields), ") = ("))
             is_first_value = True
             index = 1
             for value, is_raw_value in self.__fields.values():
                 if not is_first_value:
                     sql.append(", ")
-                is_first_value = False
-                if not is_raw_value:
-                    sql.append(f"${argument_count + index}")
+                if is_raw_value:
+                    sql.append(str(value))
+                else:
+                    sql.append(f"${len(arguments) + index}")
```

Comment:
Please, correct me if I'm wrong but I believe this isn't an equivalent transformation, because removing `argument_count` and using the actual length of the `arguments` list - which is constantly modified - will result in the wrong argument indices being calculated. You may still be able to avoid it if you drop `index` and add +1 instead (?), but it requires some more thought.